[허소정][Feature/products] 상품 모델, 엔드포인트 수정

sojeong20 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 상품 모델 수정(이미지 테이블 분리)
- 상품 디테일 엔드포인트 수정
- 상품 리스트 엔드포인트 수정
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)


## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

sojeong20 committed:
views.py 수정중

sojeong20 committed:
Merge branch 'main' into feature/products

sojeong20 committed:
Add: 상품 디테일 뷰 생성
- products/views.py ProductDetailView 생성
- products/models.py color field Null=True 속성 추가
- URL path 추가

sojeong20 committed:
Modify: 상품 상세페이지 엔드포인트 수정

sojeong20 committed:
상품디테일뷰 수정

sojeong20 committed:
product_info 딕셔너리 형태로 변환

sojeong20 committed:
product_info 수정

sojeong20 committed:
Modify: ProductDetailView refactoring

sojeong20 committed:
Modify: path parameter 수정, 변수명 변경

soheon-lee commented on products/views.py:
```diff
@@ -1,3 +1,35 @@
+            images = product.productimage_set.filter(color_id=color_id)
+            product_images = [image.image_url for image in images]
+            
+            if color_id:
+                color_ids = product.productimage_set.values_list('color_id',flat=True).distinct()
+                color_options = [{"id": id, "name": Color.objects.get(id=id).name} for id in color_ids]
+            else:
+                color_options = []
```
이처럼 내부 속성을 골라서 꺼내와야 하거나, 있을 수도 없을 수도 있는 속성을 관리할 때에는 `values`나 `values_list`보다는 `all()`메소드를 사용하여 객체를 모두 불러온 후 객체의 속성을 다루어 딕셔너리를 생성하시는 것이 훨씬 직관적입니다 :) 이따 거실에서 잠깐 뵐게요!

sojeong20 committed:
Add: 상품 리스트 엔드포인트 생성
- ProductListView 생성
- url 수정

sojeong20 committed:
Modify: 모델 수정(image table 분리)

sojeong20 committed:
Modify : 상품리스트, 상품디테일 엔드포인트 수정
- 상품리스트 color option 추가
- 상품리스트 쿼리스트링 id에서 이름으로 변경
- 상품디테일 pair with, 추천상품 리스트 추가

sojeong20 committed:
Modify: prefetch_related, select_related 추가

sojeong20 committed:
Fix: colorname empty string if None

jamessoun93 reviewed (changes requested):
PENDING

sojeong20 committed:
conflict 해결

sojeong20 committed:
conflict 해결

soheon-lee commented:
LGTM
